src/widgets/two-list.ts

```diff
@@ -0,0 +1,123 @@
+    }
+
+    public async getNumberOfVisibleItems(): Promise<number> {
+        return (await this.getAllVisibleElemItems()).count();
+    }
+
+    public async getNumberOfSelectedVisibleItems(): Promise<number> {
+        return (await this.getSelectedVisibleElemItems()).count();
+    }
+
+    public async getAllAvailableItems(): Promise<string[]> {
+        const itemList: string[] = [];
+        const numberOfAvailableItems = await this.getNumberOfAvailableItems();
+        for (let position = 1; position <= numberOfAvailableItems; position++) {
+            itemList.push(await(this.byCSS(`[id^=available${position - 1}]`)).getText());
+        }
+        return itemList;
+    }
+
+    public async getAllVisibleItems(): Promise<string[]> {
+        const itemList: string[] = [];
+        const elemItems = await this.getAllVisibleElemItems();
+        const numberOfVisibleItems = await elemItems.count();
+        for (let position = 1; position <= numberOfVisibleItems; position++) {
+            itemList.push(await elemItems.get(position - 1).getText());
+        }
+        return itemList;
+    }
+
+    public async getSelectedAvailableItemPositions(): Promise<number[]> {
+        const positionList: number[] = [];
+        const elemItems = await this.getSelectedAvailableElemItems();
+        const numberOfSelectedItems = await elemItems.count();
+        for (let index = 0; index < numberOfSelectedItems; index++) {
+            let positionOfSelectedItem = parseInt((await elemItems.get(index).getAttribute('id')).replace('available', '')) + 1;
+            positionList.push(positionOfSelectedItem);
+        }
+        return positionList;
+    }
+
+    public async getSelectedVisibleItemPositions(): Promise<number[]> {
+        const positionList: number[] = [];
+        const elemItems = await this.getSelectedVisibleElemItems();
+        const numberOfSelectedItems = await elemItems.count();
+        for (let index = 0; index < numberOfSelectedItems; index++) {
+            let positionOfSelectedItem = parseInt((await elemItems.get(index).getAttribute('id')).replace('element', '')) + 1;
+            positionList.push(positionOfSelectedItem);
+        }
+        return positionList;
+    }
+
+    private async getAllVisibleElemItems(): Promise<ElementArrayFinder> {
+        return this.allByCSS('div[class~="list-group"] > div[class~="slab-sortable-list-row"]');
+    }
+
+    private async getAllAvailableElemItems(): Promise<ElementArrayFinder> {
+        return this.allByCSS('li[id^=available]');
+    }
+
+    private async getSelectedVisibleElemItems(): Promise<ElementArrayFinder> {
+        return this.allByCSS('div[class~="list-group"] > div[class~="slab-sortable-list-row-selected"]');
+    }
+
+    private async getSelectedAvailableElemItems(): Promise<ElementArrayFinder> {
+        return this.allByCSS('li[class~="slab-twolistboxrowselected"]');
+    }
+
+    public getRemoveAllLink(): Link {
```

Comment:
I would move up this two public methods above the private ones

Reply:
Sure. Resorted